Import History

blackforestboi opened the pull request:
i made some changes, now at least we have the url_html parsed into a real DOM tree. this should help.

blackforestboi committed:
HTML to dom

bohrium272 reviewed (changes requested):
Please change the loop statement at line 372 to `(0, history.length)`

blackforestboi committed:
Extended Limit for chrome history api search to 200.000 - Works

half-fixed the issue with cluttered html request. it saves now, but still it is not the most beautiful code that is saved.

blackforestboi committed:
changed loop statement.

oliversauter commented:
oh yes! had that for experimental purposes. Thanks for correcting. Fixed!

blackforestboi committed:
added the check for blacklist to `saveHistory` message.

blackforestboi committed:
onboarding process finished! :)

what is missing is that we delete elements from the json file that have been processed. this way people can restart the whole process of importing, in case it fails.

blackforestboi committed:
Sooo now i finished up the whole thing :)

Its now also possible to cancel the download and resume it (it remembers the elements that already have been downloaded).

Practially the feature of importing your history is finished.

blackforestboi committed:
added standard blacklist websites.

blackforestboi committed:
Changed Banner location to local

blackforestboi committed:
remove button, not make it invisible

obsidianart commented on extension/js/download_History.js:
```diff
@@ -0,0 +1,114 @@
+                document.getElementById("title_download").innerHTML = "Download cancelled!";
+                document.getElementById("close_message").innerHTML = "You can always restart it via the settings.";
+                document.getElementById("info_text").innerHTML = '<a href="preferences.html">Go to settings</a>';
+                document.getElementById('abort_button').remove()
```
to do it properly you need to remove the click event first or you will create memory leak

blackforestboi replied:
I don't understand that one.

obsidianart replied:
at line 8 you have

```
document.getElementById('abort_button').onclick = function(){
    abort_status = 1
};
```

which look like a bind but I just realized is not (never saw this tecnique before), you can just ignore this comment

obsidianart commented on extension/js/background.js:
```diff
@@ -123,6 +123,7 @@ function handleMessage(data, sender, sendRespones) {
     // data is from message
     if (data.msg === 'pageContent' && shouldArchive(data)) {
         delete data.msg;
+        console.log(data.text)
```
needed?

blackforestboi replied:
removed

obsidianart commented on extension/js/download_History.js:
```diff
@@ -0,0 +1,114 @@
+
+//open list of already indexed urls from local storage
+var existing_urls = JSON.parse(localStorage['list_downloaded_urls']);
+var abort_status = 0
```
if is only 0 or 1 better to say isAbortedByUser = false

blackforestboi replied:
done.

obsidianart commented on extension/assets/download_history.html:
```diff
@@ -0,0 +1,34 @@
+  <p id="info_text">To make all your previously visited pages available for full-text search, we need to download them first. <br>Depending on the number of pages, this might take a while and might influence your browsing speed.<br> <strong>To cancel the download, deactivate and activate the WorldBrain extension.</strong></p>
+
+
+<button id="abort_button" style="float:right" class="ui red button">Cancel Download</button>
```
I would say this stop downloading, doesn't cancel it as you are saving what you downloading till that point

blackforestboi replied:
legacy piece. Forgot to remove it. was before I added the cancel button. Thanks!

obsidianart commented on extension/js/background.js:
```diff
@@ -135,15 +136,15 @@ function handleMessage(data, sender, sendRespones) {
         timeIndex.push(time.toString());
```
I don't see the point of changing time toString(), numbers are as good (maybe there's a reason I can't understand)

obsidianart commented on extension/js/download_History.js:
```diff
@@ -0,0 +1,114 @@
+                document.getElementById("close_message").innerHTML = "You can always restart it via the settings.";
+                document.getElementById("info_text").innerHTML = '<a href="preferences.html">Go to settings</a>';
+                document.getElementById('abort_button').remove()
+                console.log("Cancelled Download")
```
"Downloaded cancelled" (but I would say is stopped not cancelled)

blackforestboi replied:
Done

obsidianart commented on extension/js/download_History.js:
```diff
@@ -0,0 +1,114 @@
+                        // for this whole thing we need a better solution because many divs actually have the same textContent and 
+                        //therefore we have things multiple times -> unecessary bloating of file, now ca 150kb, could be 30kb. 
+                        body_text = doc.body
+                        divs = body_text.getElementsByTagName('div')
```
are you assuming the content is inside div? what about article, section, main and so on?

obsidianart replied:
We should go in this direction or find an open version of this https://www.readability.com/developers/api/parser

blackforestboi replied:
Actually we use readablity already for parsing the article before saving it. Will test out if we can use the local version we have. 

@arpitgogia said it only return parts of the text but that was before the adjustments to the correct DOM tree of the HTML file.

blackforestboi replied:
readabilities library is also not ideal. The browser itself with document.body.textContent is the best way to retrieve the content.

I have not found a way to extract just the visible textual content from an article with a JS library.

blackforestboi committed:
fixes to @obsidianart's suggestions.

Missing the parsing via readablity yet. Comes soon!

blackforestboi committed:
got the readability parser now completely running. Filesize before: 111kb now: 8kb
:)

blackforestboi committed:
updates about variants of crawling the website

blackforestboi committed:
Change in what is stored.

blackforestboi committed:
Added Automatical reload of plugin, so that newly imported urls are taken into index.

oliversauter commented:
closing as being obsolete with #27
